modP2pImpl/src/org/aion/p2p/impl/comm/Node.java

```diff
@@ -80,7 +81,7 @@
         this.idHash = 0;
         this.ip = ipStrToBytes(_ipStr);
         this.ipStr = _ipStr;
-        this.port = port;
+        this.port = ((port > 65535 || port < 1) ? 0 : port);
```

Comment:
Should we just throw an exception instead of returning 0?  I don't think the rest of the code will work if port is set to 0.  Or at least log an error/warning